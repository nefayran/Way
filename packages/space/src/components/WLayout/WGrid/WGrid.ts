import { computed, h, defineComponent } from "vue";

export default defineComponent({
  name: "WGrid",

  props: {
    cols: {
      type: [Boolean, String, Number],
      default: false,
    },
    rows: {
      type: [Boolean, String, Number],
      default: false,
    },
    tag: {
      type: String,
      default: "div",
    },
  },

  setup(props, { slots }) {
    const classes = computed(() => {
      const classList: any[] = [];

      classList.push({
        [`cols-${props.cols}`]: props.cols,
        [`rows-${props.rows}`]: props.rows,
      });

      return classList;
    });

    return () =>
      h(
        props.tag,
        {
          class: ["w-grid grid", classes.value],
        },
        slots.default?.()
      );
  },
});
