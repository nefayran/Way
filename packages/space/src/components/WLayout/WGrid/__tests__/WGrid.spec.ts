import { mount } from "@vue/test-utils";
import WGrid from "../WGrid";

describe("WGrid.vue", () => {
  test("should have an w-grid class", () => {
    const wrapper = mount(WGrid, {
      propsData: {
        cols: "6",
      },
    });
    expect(wrapper.classes()).toContain("w-grid");
    expect(wrapper.html()).toMatchSnapshot();
  });
});
