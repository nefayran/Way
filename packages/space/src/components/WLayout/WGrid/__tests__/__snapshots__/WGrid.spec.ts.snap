// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WGrid.vue should have an w-grid class 1`] = `<div class="w-grid grid cols-6"></div>`;
